### PR TITLE
Fix: crm_mon: detect when run on remote-node

### PR DESCRIPTION
**Pull request by wenningerk**
Looks like a clean solution so far.
We had EREMOTEIO already that can be used to notify that there is pacemaker-remoted but the api doesn't support proxying.
Have to check if I can make the proxy-callback NULL but accept.

### CODE REVIEW

File: lib/common/ipc_server.c [hunk 1]
```diff
@@ -921,6 +921,27 @@ pcmk__serve_fenced_ipc(qb_ipcs_service_t **ipcs,
...
+void
+pcmk__serve_pacemakerd_ipc(qb_ipcs_service_t **ipcs,
+                       struct qb_ipcs_service_handlers *cb)
+{
+    *ipcs = mainloop_add_ipc_server(CRM_SYSTEM_MCP, QB_IPC_NATIVE, cb);
+
+    if (*ipcs == NULL) {
+        crm_err("Couldn't start pacemakerd IPC server");
+        crm_warn("Verify pacemaker and pacemaker_remote are not both enabled.");
+        crm_exit(CRM_EX_OSERR);
```
Comment (kgaillot) [inline lib/common/ipc_server.c:941]: The other similar functions all return CRM_EX_FATAL, which makes sense to me -- it's unlikely to succeed if respawned. I don't think anything is relying on the OSERR code.

File: tools/crm_mon.c [hunk 2]
```diff
@@ -988,48 +988,60 @@ pacemakerd_status(void)
...
                     }
                 }
             }
-        }
+            break;
+        case EREMOTEIO:
+            rc = pcmk_rc_ok;
+            if ((output_format == mon_output_console) ||
+                        (output_format == mon_output_plain)) {
+                print_as(output_format, "Running on remote-node ...\n");
```
Comment (kgaillot) [inline tools/crm_mon.c:1039]: I think we can skip a message here to avoid clutter, since we're not actually doing anything

**Review by kgaillot — COMMENTED** [review]
Good solution

File: lib/common/ipc_server.c (hunk 1 above)
Comment (wenningerk) [reply lib/common/ipc_server.c:941]: But there is anyway nobody there to handle that restart-logic - right.
So I thought stay with where we were before for now.

File: tools/crm_mon.c (hunk 2 above)
Comment (wenningerk) [reply tools/crm_mon.c:1039]: hmm ... actually was quite nice in my tests as I could see the difference between nothing running at all and cluster not connected to pacemaker-remote.

Comment (wenningerk) [reply tools/crm_mon.c:1039]: maybe limit it to curses-console ...

File: lib/common/ipc_server.c (hunk 1 above)
Comment (kgaillot) [reply lib/common/ipc_server.c:941]: True, at this point it's better to keep the behavior -- let's just add a comment why it's different from all the other functions.

File: tools/crm_mon.c (hunk 2 above)
Comment (kgaillot) [reply tools/crm_mon.c:1039]: Sounds good, let's limit it to console and reword it to something like "Waiting for cluster connection to this remote node" or whatnot. For text, maybe we use a similar message for the error output (maybe via the GError).

Comment (wenningerk) [reply tools/crm_mon.c]: flag on_remote_node is clumsy but I guess with the overhaul that makes print_as go away this can as well go again.

File: lib/common/ipc_server.c (hunk 1 above)
Comment (wenningerk) [reply lib/common/ipc_server.c:946]: yep - put in some explanatory comment
